packages/block-library/src/template-part/variations.js

```diff
@@ -38,9 +24,33 @@ const variations = [
 			"The footer template defines a page area that typically contains site credits, social links, or any other combination of blocks. Since it's a global element it can be present across all pages and posts."
 		),
 		icon: footer,
-		isActive: createIsActiveBasedOnArea( 'footer' ),
-		scope: [],
+		attributes: { area: 'footer' },
+		scope: [ 'inserter' ],
 	},
 ];
 
+/**
+ * Add `isActive` function to all `Template Part` variations, if not defined.
+ * `isActive` function is used to find a variation match from a created
+ *  Block by providing its attributes.
+ */
+variations.forEach( ( variation ) => {
+	if ( variation.isActive ) return;
+	variation.isActive = ( blockAttributes, variationAttributes ) => {
+		const { area, theme, slug } = blockAttributes;
+		// We first check the `area` block attribute which is set during insertion.
+		// This property is removed on the creation of a template part.
+		if ( area ) return area === variationAttributes.area;
```

Comment:
I like this flow, it should make a lot of things more flexible for full template patterns